internal/thelma/utils/repeater/repeater_test.go

```diff
@@ -0,0 +1,100 @@
+package repeater
+
+import (
+	"github.com/stretchr/testify/assert"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestNew_panics(t *testing.T) {
+	assert.Panics(t, func() {
+		New(nil)
+	})
+}
+
+func TestRepeater_startIntervalStop(t *testing.T) {
+	var executions int
+	var executionsMutex sync.Mutex
+	run := func() {
+		executionsMutex.Lock()
+		executions++
+		executionsMutex.Unlock()
+	}
+	r := New(run, func(o *Options) {
+		o.Interval = time.Second / 3
```

Comment:
Another way to do these tests that would avoid the sleeps could be to define a run function that pushes items into an output channel and then assert that the receiving side is what is expected.

Reply:
Yeah, my other thought was that I could just not sleep at all and let the mutex guard it, but since the interval is such a core part of what this package does I figured it would make sense to actually test that the interval was happening. Maybe in other calling packages then we can go with the channel or count route since the delay would be already covered